music_this_week_app/views.py

```diff
@@ -79,23 +80,35 @@ def callback(request):
 
 def setup(request):
     """Displays search args and a search button, which links to /search"""
-
+    #get user ip address to find user location 
+    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR')
+    if x_forwarded_for:
+        ip = x_forwarded_for.split(',')[0]
+    else:
+        ip = request.META.get('REMOTE_ADDR')
+    #handle invalid IP or ip == 127.0.0.1 because running locally 
+    try:
+        g = GeoIP2()
+        location = g.city(ip)
+    except AddressNotFoundError:
+        location = {'city' : "San Francisco"}
     # Fetch user data
     pc = request.session.get('pc')
     if pc is None:
         print("ERROR: setup called without pc session. Redirecting home.")
         return HttpResponseRedirect('/')
 
-    context = pc.user_info
+    context = dict(pc.user_info, **{'location' : location['city']}) 
     return render(request, 'music_this_week/setup.html', context)
 
 def search(request):
     """ Search for shows and create playlist then redirect to it"""
     # Parse request for search arguments
     search_args = dict(request.GET.items())
-
+    print(search_args)
```

Comment:
If you want this in the log statements, please add some plain text describing what it is so it doesn't just show up as a random dictionary:

`{u'latlng_checkbox': u'on', u'nResults': u'200', u'hidden_lat': u'37.7635279', u'location': u'San Francisco', u'date': u'Next 7 Days', u'hidden_lon': u'-122.42203899999998'}`

Something like `"Search arguments:"` would be fine.

Though it's a bit confusing that both the `location` and `hidden_lat/lon` are sent if one of them is not used. I need to dig a bit deeper to Identify where they are filtered out.